src/client/components/pages/Courses/RoomSelectionTable.tsx

```diff
@@ -0,0 +1,162 @@
+import React, { ReactElement, useState, ChangeEvent } from 'react';
+import styled from 'styled-components';
+import {
+  Table,
+  TableRow,
+  TableHead,
+  TableHeadingCell,
+  TableCell,
+  TableBody,
+  VARIANT,
+  Button,
+  LoadSpinner,
+  Dropdown,
+  TableRowHeadingCell,
+  fromTheme,
+} from 'mark-one';
+import RoomResponse from 'common/dto/room/RoomResponse.dto';
+
+interface RoomSelectionTableProps {
+  /** Whether an asynchronous request to the server has been made */
+  dataFetching?: boolean;
+  /** The list of rooms to show in the list */
+  roomList: RoomResponse[];
+  /** A handler to be called when the add button is clicked */
+  addButtonHandler: (roomData: RoomResponse) => void;
+}
+
+/**
+ * The allowed values for Availability filter in the room table
+ */
+
+enum AVAILABILITY {
+  ALL='All',
+  AVAILABLE='Available',
+  UNAVAILABLE='Unavailable',
+  CHECK='Check FAS availability'
+}
+
+/**
+ * A textbox that will appear before a meeting day/time has been selected
+ */
+
+const RoomSelectionTablePrompt = styled.div`
+  border: ${fromTheme('border', 'light')};
+  border-top: none;
+  text-align: center;
+  font-weight: ${fromTheme('font', 'bold', 'weight')};
+  font-size: ${fromTheme('font', 'bold', 'size')};
+  font-family: ${fromTheme('font', 'bold', 'family')};
+  padding: ${fromTheme('ws', 'medium')};
+`;
+
+/**
+ * Formats the meeting data for the "Availability" column
+ */
+const displayAvailability = (roomData: RoomResponse) => {
+  const { campus, meetingTitles } = roomData;
+  if (campus === 'FAS') {
```

Comment:
I think this came out of one of our demos for the users. There are cases where SEAS courses can be held in FAS rooms (e.g., the Science Center), but the course-planner can't track all of the FAS courses to tell if those rooms are free or not. So the workaround (reflected in [the mockups](https://seas-computing.github.io/course-planner_ux/courses.html)) is just to flag those rooms for the user to manually check availability.

re: an enum, that could make sense. However, `campus` is currently its own database table to allow for future flexibility, and if we were using an enum we'd probably end up needing to make that an enum column in the database as well. We may well do that, but I think it's a little out of scope for this ticket. I suggest we discuss at the sprint retrospective and add a future ticket if needed.

Reply:
That's a good idea. I had the same thought of converting the table to an enum - however as you point out, that would be out of scope of the ticket. It would also potentially introduce maintainability issues.